test/monitoring_console/monitoring_console_test.go

```diff
@@ -492,4 +535,67 @@ var _ = Describe("Monitoring Console test", func() {
 			}
 		})
 	})
+
+	Context("Multisite Clustered deployment (M4 - 3 Site clustered indexer, search head cluster)", func() {
+		It("monitoring_console, integration: MC can configure SHC, indexer instances after scale up and standalone in a namespace", func() {
+			/*
+				Test Steps
+				1. Deploy Multisite Indexer Cluster
+				2. Deploy Monitoring Console
+				3. Wait for Monitoring Console status to go back to READY
+				4. Verify SH are configured in MC Config Map
+				5. Verify SH are configured in peer strings on MC Pod
+				6. Verify Indexers are configured in MC Config Map
+				7. Verify Monitoring Console Pod has peers in Peer string on MC Pod
+			*/
+			defaultSHReplicas := 3
+			defaultIndexerReplicas := 1
+			siteCount := 3
+			mcName := deployment.GetName()
+			err := deployment.DeployMultisiteClusterWithMonitoringConsole(deployment.GetName(), defaultIndexerReplicas, siteCount, mcName, true)
+			Expect(err).To(Succeed(), "Unable to deploy Cluster Master")
+
+			// Ensure that the cluster-master goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure indexers go to Ready phase
+			testenv.IndexersReady(deployment, testenvInstance, siteCount)
+
+			// Ensure indexer clustered is configured as multisite
+			testenv.IndexerClusterMultisiteStatus(deployment, testenvInstance, siteCount)
+
+			// Ensure search head cluster go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Deploy Monitoring Console CRD
+			mc, err := deployment.DeployMonitoringConsole(deployment.GetName(), "")
+			Expect(err).To(Succeed(), "Unable to deploy Monitoring Console instance")
+
+			// Verify Monitoring Console is Ready and stays in ready state
+			testenv.VerifyMonitoringConsoleReady(deployment, deployment.GetName(), mc, testenvInstance)
+
+			// Get contents of MC config map
+			mcConfigMap, err := testenv.GetMCConfigMap(deployment, testenvInstance.GetName(), mcName)
+			Expect(err).To(Succeed(), "Unable to get MC config map")
+
+			// Check Monitoring console is configured with all search head instances in namespace
```

Comment:
Should we also check if cluster master is present in configured peers list?

Reply:
The CM part if checked indirectly when we verify Indexers